automation/automation/jsenv_handler.go

```diff
@@ -0,0 +1,66 @@
+package automation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cortezaproject/corteza-server/pkg/jsenv"
+)
+
+type (
+	jsenvHandler struct {
+		reg queueHandlerRegistry
+	}
+)
+
+func JsenvHandler(reg queueHandlerRegistry) *jsenvHandler {
+	h := &jsenvHandler{
+		reg: reg,
+	}
+
+	h.register()
+	return h
+}
+
+func (h jsenvHandler) process(ctx context.Context, args *jsenvProcessArgs) (res *jsenvProcessResults, err error) {
+	res = &jsenvProcessResults{}
+
+	if !args.hasFunction {
+		err = fmt.Errorf("could not process payload, function missing")
+		return
+	}
+
+	if !args.hasScope {
+		err = fmt.Errorf("could not process payload, scope missing")
+		return
+	}
+
+	// call jsenv, feed it function and expect a result
+	tr := jsenv.NewTransformer(jsenv.LoaderJS, jsenv.TargetNoop)
+	vm := jsenv.New(tr)
+
+	fn, err := vm.RegisterFunction(args.Function)
+
+	if err != nil {
+		err = fmt.Errorf("could not register jsenv function: %s", err)
+		return
+	}
+
+	out, err := fn.Exec(vm.New(args.scopeString))
+
+	if err != nil {
+		err = fmt.Errorf("could not exec jsenv function: %s", err)
+		return
+	}
+
+	switch vv := out.(type) {
+	case uint64, int64:
+		res.ResultInt = int64(vv.(uint64))
```

Comment:
can you just add a quick comment to these two that they might be removed (if I understand you correctly, this is the case here, right?)

Reply:
:+1: